application/src/main/java/uk/nhs/digital/uec/api/filter/LocalAccessTokenFilter.java

```diff
@@ -0,0 +1,22 @@
+package uk.nhs.digital.uec.api.filter;
+
+import java.io.IOException;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@Component
+@Profile("local")
+public class LocalAccessTokenFilter extends OncePerRequestFilter {
+
+  @Override
+  public void doFilterInternal(
+      HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+      throws ServletException, IOException {
+    filterChain.doFilter(request, response);
```

Comment:
We should have an implementation that will check if its received the test token at least so that the application isn't left completely open.

Reply:
The local filter completely skips the access token JWT validation as the local request can contain any format of access token, which will not match the jwt 3 part token structure

Reply:
There has to be a way to extract the token from the request in this method.  As you say Arif, this won't work with the JWT utils, but there has to be a way to extract the raw token as a string and then to just perform a comparison. For example, if we set the token in the GET request to be 'SERVICE_FINDER_MOCK_TOKEN', we should be able to extract that string from the request and check that it equals 'SERVICE_FINDER_MOCK_TOKEN'.
We can swarm on this part tomorrow if that's easier.